modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/consistency/GridNearReadRepairCheckOnlyFuture.java

```diff
@@ -125,6 +146,8 @@ public GridNearReadRepairCheckOnlyFuture(
      * @return Future represents 1 entry's value.
      */
     public <K, V> IgniteInternalFuture<V> single() {
+        init();
```

Comment:
What if we add a parameter with an initialization flag for the constructor `GridNearReadRepairAbstractFuture`? Now it looks strange and a little bit dangerous (it's possible to forget to `init()` future somewhere)